### ISSUE TITLE
Custom Breaks

### ISSUE THREAD

**Issue by bduff9** [hovancik/stretchly#177]
Hey there!  Love the app, its the best one I've used to date and I can definitely see it benefiting me after only using for a week so far.

One minor recommendation is similar to #161 for a snooze button.  For me, I kind of agree with you about never taking breaks if it can be snoozed, however one thing I have noticed is that when the micro break happens, I always rest.  However, I end up skipping the majority of breaks because its not a good stopping point.  Instead, I find myself taking breaks outside of the scheduled ones for bathroom breaks, lunch, talking with coworkers, etc.

Therefore, it might be nice to be able to tell Stretchly that I am starting and then stopping a break.  It can then see how long I took, and take that into account for its schedule.  Otherwise, there have been a couple times where I came back and it then told me to break pretty quickly after.  I can always skip the regular breaks but I turned off the skip for micro breaks to force myself to take them.  I also don't necessarily know how long I took on the break.

I would see this as a simple menu option with a `Start Break` option.  Once clicked, maybe it blocks the screen(s) (or just shows a dialog so the computer could still be used) with only a `Stop Break` option.  Stretchly could then see how long was taken and if its closer to a micro break or break, count it as such in the schedule.

I hope this is clear enough, however if not, please let me know and I will elaborate.  What do you think about such a feature at some point in the future?  Thanks again for your work!

**Comment by hovancik** [comment]
Hi,  what platform are you on? And do you have idle time detection on?  

Idle time should take care of that: when your machine is idle for more then 5 minutes, it stops breaks until you come back and then reset timers.

**Comment by bduff9** [comment]
Hey, thanks for your reply.  I am on Windows 10, and yes, I turned on idle time when I first installed stretchly.

Maybe it is not working for me then as I am constantly getting interrupted while working, even though I just took a break previously.  On the other hand, it is possible that whatever metric you are using for idle time won't work for me because, as a software engineer, I constantly have things running (code compiling, test suites testing, etc.) so even when I walk away, the CPU is rarely completely idle.

I also sometimes go to the bathroom or for a chat and come back after only 3 or 4 minutes, but I would at least like the option to have that count as a break as if I break for 4 minutes and come back, I don't want to take another 5 minute break right after.

I suppose if the idle time could be improved, that might be good enough, but as I said, I'm not sure if its working on my system or maybe just not working for my specific use case.
